### PR TITLE
fix(table): Various bugs

### LINKED ISSUE #8155: [tables] Usability tracking issue
- [x] changing column format causes crash (composer only!?)

  https://github.com/user-attachments/assets/f79e4107-ad93-4d8c-8f22-e75bc5851a4e

- [x] Sidebar overlaps select:
    

- [ ] Click column action menu > click column settings > click type select > click somewhere else in the UI ... UI disabled?

  https://github.com/user-attachments/assets/37fbcfe7-2098-4632-b40c-7b7e85387297

- [x] Table not loading in composer on create or page reload (sometimes)

### PR DESCRIPTION
**Pull request by ZaymonFC**
- Fix parsing and formatting for scalar types
- Support adding columns to table
- Delete and restore fields using ViewProjection
- Tables not loading correctly in composer on create or page reload
- Resolves #8155

### CODE REVIEW

File: packages/plugins/plugin-table/src/TablePlugin.tsx [hunk 1]
```diff
@@ -150,45 +151,26 @@ export const TablePlugin = (): PluginDefinition<TablePluginProvides> => {
...
-              if (!intent.undo) {
-                const fieldPosition = table.view.fields.indexOf(field);
-                if (fieldPosition === undefined) {
-                  return;
-                }
-
-                // TODO(ZaymonFC): We need to manipulate the schema with another method here.
-                // removeFieldFromView(table.schema, table.view, field);
+              const schema = getSpace(table)?.db.schemaRegistry.getSchema(table.view.query.__typename);
+              invariant(schema, 'Table schema not found');
```
Comment (richburdon) [inline packages/plugins/plugin-table/src/TablePlugin.tsx:160]: are these messages helpful? if so, add type at end of message and keep terse.
`Schema not found: ${}`

File: packages/plugins/plugin-table/src/components/Table/Table.tsx [hunk 2]
```diff
@@ -66,7 +65,7 @@ const TablePrimitive = ({ model }: TableProps) => {
   const { state: menuState, triggerRef, handleClick, close, showColumnSettings } = useTableMenuController();
 
   return (
-    <ModalPrimitive.Root>
+    <>
       {/* TODO(burdon): Is this required to be unique? */}
       <Grid.Root id='table-next'>
```
Comment (richburdon) [inline packages/plugins/plugin-table/src/components/Table/Table.tsx:70]: is this id required?

File: packages/plugins/plugin-table/src/hooks/useTableIntialisation.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { type TableType } from '../types';
+
+/**
+ * Initializes a new view for table if one does not exist.
+ * Creates a new schema and registers it with the database.
+ */
+export const useTableIntialisation = (table?: TableType) => {
+  const space = getSpace(table);
+
+  useEffect(() => {
```
Comment (richburdon) [inline packages/plugins/plugin-table/src/hooks/useTableIntialisation.ts:21]: i think you like this pattern :), but i feel it's clearer to provide a general purpose, testable function and wrap with an effect where it is needed.

```
useEffect(() => {
  if (table && table?.view) {
    initTable(space, table);
  }
}, [space, table])
```

File: packages/plugins/plugin-table/src/hooks/useTableModel.ts [hunk 4]
```diff
@@ -62,7 +31,7 @@ export const useTableModel = ({
 
     let tableModel: TableModel | undefined;
```
Comment (richburdon) [inline packages/plugins/plugin-table/src/hooks/useTableModel.ts:32]: i think it's okay to redefined model in this scope.

```
let model = 
```

File: packages/sdk/schema/src/formatting.ts [hunk 5]
```diff
@@ -18,99 +18,103 @@ type ValueFormatProps = {
 // TODO(burdon): Move to react-ui-data.
```
Comment (richburdon) [inline packages/sdk/schema/src/formatting.ts:18]: execute this TODO to share with sheet?

File: packages/sdk/schema/src/types/projection.test.ts [hunk 6]
```diff
@@ -167,5 +168,89 @@ describe('ViewProjection', () => {
...
+    const emailIndex = initialFieldsOrder.indexOf('email' as JsonPath);
+    const initialEmail = projection.getFieldProjection('email');
+    const initialSchemaProps = { ...mutable.jsonSchema.properties! };
+
+    // Delete and restore
+    const { deleted, index } = projection.deleteFieldProjection('email');
+
+    console.log('deleted', deleted);
+
+    // Verify email is deleted but name is unchanged
```
Comment (richburdon) [inline packages/sdk/schema/src/types/projection.test.ts:237]: nit: standard is to terminate with a period.

**Review by richburdon — APPROVED** [review]

File: packages/plugins/plugin-table/src/components/Table/Table.tsx (hunk 2 above)
Comment (ZaymonFC) [reply packages/plugins/plugin-table/src/components/Table/Table.tsx:70]: Yes, not sure why though.

**Pull request merged**